packages/webapp/src/components/RepeatCropPlan/index.jsx

```diff
@@ -0,0 +1,387 @@
+                <DaysOfWeekSelect
+                  onChange={onChange}
+                  defaultValue={value}
+                  maxSelect={1}
+                  errors={getInputErrors(errors, DAYS_OF_WEEK)}
+                />
+              )}
+            />
+          </div>
+        )}
+
+        {repeatInterval?.value === 'month' && (
+          <div className={styles.monthlyOptionsSelect}>
+            <Controller
+              control={control}
+              name={MONTH_REPEAT_ON}
+              rules={{ required: true }}
+              render={({ field: { onChange, value } }) => (
+                <ReactSelect
+                  options={monthlyOptions}
+                  onChange={(e) => {
+                    onChange(e);
+                  }}
+                  value={value}
+                />
+              )}
+            />
+          </div>
+        )}
+
+        <div>
+          <Label className={styles.label}>{t('REPEAT_PLAN.FINISH')}</Label>
+
+          <RadioGroup
+            hookFormControl={control}
+            name="finish"
+            required
+            onChange={() => {
+              trigger(FINISH_ON_DATE);
+            }}
+            radios={[
+              {
+                label: '',
+                value: 'after',
+                children: (
+                  <div className={styles.radioGroup}>
+                    <p>{t('REPEAT_PLAN.AFTER')}</p>
+                    <Input
+                      type="number"
+                      stepper
+                      onKeyDown={integerOnKeyDown}
+                      min={1}
+                      max={20}
+                      hookFormRegister={register('after_occurrences', {
+                        validate: (value) => {
+                          if (!value && getValues(FINISH) === 'after') {
+                            return t('common:REQUIRED');
+                          }
+                          return true;
+                        },
+                      })}
+                      style={{ width: '72px' }}
+                      onChange={() => {
+                        setValue(FINISH, 'after');
+                        trigger(FINISH_ON_DATE);
+                      }}
+                    />
+                    <p>{t('REPEAT_PLAN.REPETITIONS')}</p>
+                  </div>
+                ),
+              },
+              {
+                label: '',
+                value: 'on',
+                children: (
+                  <div className={styles.radioGroup}>
+                    <p>{t('REPEAT_PLAN.ON')}</p>
+                    <Input
+                      type="date"
+                      hookFormRegister={register(FINISH_ON_DATE, {
+                        validate: validateFinishOnDate,
+                      })}
+                      onChange={() => setValue(FINISH, 'on')}
+                    />
+                  </div>
+                ),
+              },
```

Comment:
Ahh, it's because of the first interaction (`dirtyFields`) requirement; I didn't like the UX of getting an error message as soon as you pick the radio button (and before you have had a chance to interact with the date picker), but of course yes, you could submit at that point so that check needs to go!

But sorry I don't understand what you mean about the error message; if I remove the `dirtyFields` check, this is the UI as soon as you pick the "finish === 'on'". It's the same as when you clear the date picker input now. Do you not see that error message?

<img width="384" alt="Screenshot 2023-07-19 at 12 05 17 PM" src="https://github.com/LiteFarmOrg/LiteFarm/assets/82857964/a82b7ed4-3a8a-431f-bf95-0559e6024612">

Reply:
> I didn't like the UX of getting an error message as soon as you pick the radio button
I agree...

I had not tried it by removing the dirtyFields check, but I confirmed that the error message would be shown! So your solution is good :)

Reply:
Fixed! Thank you for finding that!! ❤️